sdk/rust/nym-sdk/src/mixnet/native_client.rs

```diff
@@ -198,6 +260,195 @@ pub struct MixnetClientSender {
     packet_type: Option<PacketType>,
 }
 
+impl AsyncRead for MixnetClient {
+    fn poll_read(
+        mut self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+        buf: &mut ReadBuf,
+    ) -> Poll<tokio::io::Result<()>> {
+        let mut codec = ReconstructedMessageCodec {};
+
+        if self._read.pending() {
+            return self.read_buffer_to_slice(buf, cx);
+        }
+
+        let msg = match self.as_mut().poll_next(cx) {
+            Poll::Ready(Some(msg)) => msg,
+            Poll::Ready(None) => return Poll::Ready(Ok(())),
+            Poll::Pending => return Poll::Pending,
+        };
+
+        // let mut buffer = BytesMut::new();
+
+        codec.encode(msg, &mut self._read.buffer).unwrap();
```

Comment:
is it possible for that codec to fail? 
if not, could you make a comment on why? `unwrap()` are always scary : )

Reply:
but speaking of the codec, whoever wants to use `AsyncRead` on the `MixnetClient` will get `bincode`-serialised data which might not be very desirable or expected. as an user i'd expect to receive pure mixnet bytes.

what if we had two different and configurable ways of encoding that message, "advanced" and "simple" (default). "advanced" would just do whatever is happening right now, but simple would just strip the `sender_tag` out of `ReconstructedMessage` and just propagate the raw `message : Vec<u8>` instead. what do you think?

Reply:
Yes! I've actually done the thing on the InputMessage you suggested, and added another codec that takes a Receiver as a parameter, was just thinking of how to get similar ergonomics for the ReconstructedMessage

Reply:
> is it possible for that codec to fail? if not, could you make a comment on why? `unwrap()` are always scary : )

Ouch, fixed that